docs/concepts/publishing.md

```diff
@@ -0,0 +1,87 @@
+
+```toml
+# manifest.toml
+[build.mypkg]
+command = '''
+  ...
+'''
+```
+
+```console
+$ flox publish mypkg
+```
+
+## Publish process
+
+In order to make sure that the uploaded artifact can be built reproducibly,
+Flox imposes some constraints on the build and publish process.
+
+- The Flox environment performing the build must tracked in a git repository.
+- Tracked files in the git repository must be clean.
+- The git repository has a remote defined, and the current revision has been pushed to it.
+- The Flox environment must have at least one package installed to it.
+
+All of this is there to ensure that the build environment isn't dirty and that we can associate the uploaded artifact with a point in time in the Flox Catalog.
+
+As part of the `flox publish` command, the CLI will clone the git repository to a temporary directory to ensure that any files referenced in the build are tracked by the repository.
+A clean `flox build` is then run in this directory.
+
+The package closure is then signed with the user-supplied signing key and uploaded to the organization's private catalog.
+
+## The published payload
+
+A published artifact consists of two parts:
+
+- The artifact metadata
+- The artifact itself
+
+The artifact metadata is uploaded to Flox servers so that the Flox CLI can see that it's available via the `flox search`, `flox show`, and `flox install` commands.
+The artifact itself is uploaded to the store.
+
+Today that store is user-provided, but a hosted offering will be provided in the future.
+However, when the store is user-provided it means that your organization has complete control over your artifacts, and many organizations will still choose this route even when a hosted option is available.
+
+## Consuming published artifacts
+
+Once you have uploaded an artifact via `flox publish`, the package becomes available in `flox search`, `flox show`, and `flox install`.
+To distinguish these packages from those provided by the Flox Catalog, published packages are prefixed with the name of the organization and a slash e.g. `myorg/hello`.
+
+When a user runs `flox install myorg/hello`, the artifact is downloaded directly from the store that it was published to.
+
+By default, only the owner of the private catalog has access to artifacts published to it.
+Individual users can be added to an allowlist able to access artifacts in the private catalog via the [catalog-util][catalog-util] command line tool.
+
+## Configuration
+
+### Flox Store
+
+The Flox CLI uses Nix under the hood to perform certain operations, and must be configured to be made aware of the user-provided Flox Store.
+A Flox Store is an S3-compatible service
+See the "Flox Store" cookbook page to learn more about how to provision the service.
+
+### Signing key
+
+Artifacts uploaded to a Flox Store must be signed.
```

Comment:
This is setting off alarm bells. I recall we specifically said that uploaded artifacts specifically did _not_ need to be signed, expressly so that we can put off any issues of client-side key management for this first release. Where is this coming from exactly?

Reply:
Fixed